Sources/Base/OAuth2Base.swift

```diff
@@ -451,6 +452,36 @@ open class OAuth2Base: OAuth2Securable {
 	*/
 	open func assureRefreshTokenParamsAreValid(_ params: OAuth2JSON) throws {
 	}
+	
+	
+	// MARK: - PKCE
+	
+	func generateCodeVerifier() -> String {
```

Comment:
I think these methods should be moved into `OAuth2ContextStore ` to encapsulate this implementation into the context.

Reply:
Done